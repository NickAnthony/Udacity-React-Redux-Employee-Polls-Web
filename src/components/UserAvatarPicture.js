import DefaultAvatar from "../images/DefaultAvatar.png";
import PropTypes from "prop-types";

/* This dynamically sets the avatar based on the presence of a URL. */
const UserAvatarPicture = ({ avatarURL, size }) => {
  return (
    <div
      className="profile-avatar"
      style={{
        backgroundImage: `url(${avatarURL ? avatarURL : DefaultAvatar})`,
        height: `${size ? size : 20}px`,
        width: `${size ? size : 20}px`,
      }}
    />
  );
};

UserAvatarPicture.propTypes = {
  avatarURL: PropTypes.string,
  size: PropTypes.number,
};

export default UserAvatarPicture;
