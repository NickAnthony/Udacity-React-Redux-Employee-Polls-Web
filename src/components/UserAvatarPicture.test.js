import { render } from "@testing-library/react";
import UserAvatarPicture from "./UserAvatarPicture";

describe("UserAvatarPicture", () => {
  it("will display the generic avatar icon at a size of 20px.", () => {
    var component = render(<UserAvatarPicture />);
    expect(component).toMatchSnapshot();
  });
  it("will display the provided URL at a size of 20px.", () => {
    var component = render(
      <UserAvatarPicture avatarURL={"https://i.redd.it/w6kaq2edpxha1.png"} />
    );
    expect(component).toMatchSnapshot();
  });
  it("will display the generic avatar icon at a provided size.", () => {
    var component = render(<UserAvatarPicture size={218} />);
    expect(component).toMatchSnapshot();
  });
});
