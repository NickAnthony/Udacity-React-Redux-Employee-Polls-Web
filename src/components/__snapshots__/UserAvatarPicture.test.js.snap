// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatarPicture will display the generic avatar icon at a provided size. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="profile-avatar"
        style="background-image: url(DefaultAvatar.png); height: 218px; width: 218px;"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="profile-avatar"
      style="background-image: url(DefaultAvatar.png); height: 218px; width: 218px;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UserAvatarPicture will display the generic avatar icon at a size of 20px. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="profile-avatar"
        style="background-image: url(DefaultAvatar.png); height: 20px; width: 20px;"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="profile-avatar"
      style="background-image: url(DefaultAvatar.png); height: 20px; width: 20px;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`UserAvatarPicture will display the provided URL at a size of 20px. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="profile-avatar"
        style="background-image: url(https://i.redd.it/w6kaq2edpxha1.png); height: 20px; width: 20px;"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="profile-avatar"
      style="background-image: url(https://i.redd.it/w6kaq2edpxha1.png); height: 20px; width: 20px;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
